src/core/qgspluginlayer.cpp

```diff
@@ -56,6 +56,11 @@ const QgsDataProvider *QgsPluginLayer::dataProvider() const
   return mDataProvider;
 }
 
+QIcon QgsPluginLayer::icon() const
+{
+  return QIcon();
```

Comment:
```suggestion
  return QgsIconUtils::iconForLayerType( Qgis::LayerType::Plugin );
```

Reply:
(You'll also need to add the qgsiconutils.h include)

Reply:
This leads to an endless loop，i suggest keep it